onnxruntime/test/python/onnxruntime_test_python_cudagraph.py

```diff
@@ -74,6 +117,44 @@ def testRunModelWithCudaGraph(self):  # noqa: N802
                 atol=1e-05,
             )
 
+    def testArenaWithCudaGraph(self):  # noqa: N802
+        if "CUDAExecutionProvider" in onnxrt.get_available_providers():
+            # To test cuda graph catpure, we set Arena extend strategy to be SameAsRequested so as to detect any
+            # potential memory allocation after the first run.
+            providers = [
+                ("CUDAExecutionProvider", {"enable_cuda_graph": True, "arena_extend_strategy": "kSameAsRequested"})
+            ]
+            test_model_path = get_name("squeezenet/model.onnx")
+
+            input_and_output_shape = {
+                "data_0": [16, 3, 224, 224],
+                "softmaxout_1": [16, 1000, 1, 1],
+            }
+
+            session_options = onnxrt.SessionOptions()
+            # It is optional to disable memory pattern since min_num_runs_before_cuda_graph_capture_ = 2.
+            session_options.enable_mem_pattern = False
+            session = onnxrt.InferenceSession(test_model_path, session_options, providers=providers)
+
+            cuda_graph_helper = CudaGraphHelper(session, input_and_output_shape)
+            io_binding = cuda_graph_helper.io_binding
+
+            # Create a random input for testing.
+            np.random.seed(0)
+            inputs = {"data_0": np.random.randint(0, 256, size=input_and_output_shape["data_0"]).astype(np.float32)}
+
+            # One regular run for the necessary memory allocation and cuda graph capturing
+            cuda_graph_helper.update_inputs(inputs)
+            session.run_with_iobinding(io_binding)
+            expected_output = cuda_graph_helper.get_output("softmaxout_1")
+
+            # After capturing, CUDA graph replay happens from this Run onwards
+            cuda_graph_helper.update_inputs(inputs)
+            session.run_with_iobinding(io_binding)
```

Comment:
Based on current updated code, this is the run which captures the graph and not the run which replays the graph, because minimum runs before graph capture is set to 2 and the first Run() internally is just 2 Run calls (1 actual Run + 1 recursive) which means that the third Run call is where the graph capture actually happens.